src/ts/swap.ts

```diff
@@ -217,22 +255,44 @@ export class SwapEncoder {
   public static encodeSwap(
     ...args:
       | [Swap[], Order, Signature]
+      | [Swap[], Order, Signature, Partial<SwapExecution> | undefined]
       | [TypedDataDomain, Swap[], Order, Signer, EcdsaSigningScheme]
+      | [
+          TypedDataDomain,
+          Swap[],
+          Order,
+          Signer,
+          EcdsaSigningScheme,
+          Partial<SwapExecution> | undefined,
+        ]
   ): EncodedSwap | Promise<EncodedSwap> {
-    const [domain, swaps, order, ...signatureOrSigner] =
-      args.length === 3 ? [{ name: "unused" }, ...args] : args;
-
-    const encoder = new SwapEncoder(domain);
-    encoder.encodeSwapStep(...swaps);
+    if (args.length < 5) {
+      const [swaps, order, signature, swapExecution] = args as unknown as [
+        Swap[],
+        Order,
+        Signature,
+        Partial<SwapExecution> | undefined,
+      ];
 
-    if (signatureOrSigner.length === 1) {
-      const [signature] = signatureOrSigner;
-      encoder.encodeTrade(order, signature);
+      const encoder = new SwapEncoder({});
+      encoder.encodeSwapStep(...swaps);
+      encoder.encodeTrade(order, signature, swapExecution);
```

Comment:
If `swapExecution` is undefined then the swap is guaranteed to fail for any nontrivial order, right?
Maybe in this case we should just assume that there is no extra order protection, which means that the limits are the sell and buy amounts with the right sign.

Reply:
Ah, sorry, I somehow missed it! Looks good.